app/controllers/admins/institutions_controller.rb

```diff
@@ -0,0 +1,52 @@
+class Admins::InstitutionsController < Admins::BaseController
```

Comment:
These class should be refactored to:
1. Use before action to load states, cities and find the institution
2. The flash messages should use I18n.